satpy/scene.py

```diff
@@ -1143,6 +1143,36 @@ def save_datasets(self, writer=None, filename=None, datasets=None, compute=True,
                                           **kwargs)
         return writer.save_datasets(dataarrays, compute=compute, **save_kwargs)
 
+    def compute(self, **kwargs):
+        """Call `compute` on all Scene datasets.
+
+        See :meth:`xarray.DataArray.compute` for more details.
+        """
+        new_scn = self.copy()
+        for k in new_scn._datasets.keys():
+            new_scn[k] = new_scn[k].compute(**kwargs)
```

Comment:
It would be nice if these methods could compute (same for the persist) all the DataArrays at the same time. As is this will likely recompute share dependencies of the DataArrays.

Reply:
I think I don't exactly understand what you mean with "at the same time".

Reply:
Normally for dask arrays you want to call `res1, res2, res3 = dask.array.compute(array1, array2, array3)` so that all dependency calculations for generating those three arrays are only computed once. I'm not sure how that can be done with xarray. You could try passing the DataArrays to `da.compute` and see if that works.

Reply:
Ah I see. Yes indeed that would be useful. I will check if that is possible with xarray or as you suggested with `da.compute`.